### PR TITLE
Chore: github action workflow

### PR DESCRIPTION
**Pull request by awm086**
### Changes:
- Add github actions to build, lint and test.

### Example Runs:
- https://github.com/awm086/Bodiless-JS/runs/1457298278
- https://github.com/awm086/Bodiless-JS/actions/runs/384589225

### Test Instructions: 
- Ensure Current master build and pass all tests.
- Merge into the main branch.
- Issue PR to the main branch.

### Related
Related #681

### CODE REVIEW

File: .github/workflows/build-test-lint-simple.yml [hunk 1]
```diff
@@ -1,44 +1,62 @@
...
-        env:
-          # make sure every Cypress install prints minimal information
-          CI: 1
+      - name: Print Git Version
+        run: git --version
+
+      - name: Setup git user
         run: |
-          npm ci
+          git config --global user.name "$(git --no-pager log --format=format:'%an' -n 1)"
```
Comment (dewen) [inline .github/workflows/build-test-lint-simple.yml:39]: this is to get last commit user as global git user?

Comment (awm086) [reply .github/workflows/build-test-lint-simple.yml:39]: Yes

Comment (awm086) [reply .github/workflows/build-test-lint-simple.yml:39]: It has no impact on anything except get conflict test which pulls global user. If we could remove the need for global user in the tests then we could remove this as well.

Comment (dewen) [reply .github/workflows/build-test-lint-simple.yml:39]: I mean we don't need real user here, could be fake one:
```
      - name: Setup git user
        run: |
          git config --global user.name 'gituser'
          git config --global user.email 'gituser@example.com'
```

File: .github/workflows/build-test-lint-simple.yml [hunk 2]
```diff
@@ -1,44 +1,62 @@
 name: Build, Test, Lint for Pull Requests
+env:
+  CI: true
+  FORCE_COLOR: 1
 
 on:
+  push:
```
Comment (dewen) [inline .github/workflows/build-test-lint-simple.yml:7]: Do we need this push event? There's no direct push to main branch, I think.

If in certain cases, direct push enabled, we might not need to perform the build either.

**Review by VancheeZze — APPROVED** [review]

**Review by dewen — APPROVED** [review]

**Review by hvanyo — APPROVED** [review]

**Pull request merged**